src/cnxml/glossary.js

```diff
@@ -53,13 +53,38 @@ function de_term(el, next) {
         return undefined
     }
 
+    const data = {}
+
+    if (el.parentElement && el.parentElement.tagName === 'definition') {
+        data.reference = el.getAttributeNS(
+            'http://katalysteducation.org/cmlnle/1.0', 'reference')
+    }
```

Comment:
Id suggest also checking for presence:

```js
const reference = el.getAttributeNS('http://katalysteducation.org/cmlnle/1.0', 'reference')
if (reference != null && el.parentElement && el.parentElement.tagName === 'definition') {
    data.reference = reference
}
```
this way `data.reference` will only be set if `cmlnle:reference` is actually present, and we can avoid adding `reference={null}` to all tests.